docs/src/pages/guides/FHIRValidationGuide.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: FHIR Validation Guide
+description: FHIR Validation Guide
+date:   2020-01-17 15:45:00 -0400
+permalink: /fhirvalidationguide/
+---
+
+## Overview
+
+The FHIR validation component (FHIRValidator) provides Java APIs for validating FHIR resources using constraints specified in their corresponding structure definitions. For example, in the Patient resource, we have the following constraint:
+
+```
+@Constraint(
+    id = "pat-1",
+    level = "Rule",
+    location = "Patient.contact",
+    description = "SHALL at least contain a contact's details or a reference to an organization",
+    expression = "name.exists() or telecom.exists() or address.exists() or organization.exists()"
+)
+```
+
+The validation component picks up this annotation, pulls out the FHIRPath expression and passes it on to the FHIRPath engine for evaluation. If the invariant evaluates to `false` then the FHIR validator will generate an OperationOutcome.Issue with the severity set relative to the "level" of the constraint (i.e. "Rule" or "Warning");
+
+
+## Profile Support
+
+The validation component will also validate a resource against profiles that it asserts conformance to in the `Resource.meta.profile` element assuming those profiles are available to the IBM FHIR Server via the FHIR registry component at runtime.
+
+Given a FHIR profile (structure definition) as input, the FHIR profile component generates FHIRPath expressions for a number of different types of constraints. The current scope of constraint generation is:
+
+- Cardinality constraints (required and prohibited elements)
+- Fixed value constraints (Code and Uri data types)
+- Pattern value constraints (CodeableConcept daa type)
+- Reference type constraints (FHIRPath resolve/is/conformsTo functions)
+- Extension constraints (FHIRPath `conformsTo` function)
+- Vocabulary constraints (FHIRPath `memberOf` function)
+- Choice type constraints (FHIRPath `as` function)
+
+NOTE: there is currently no support for `closed` or `ordered` slices.
+
+For example, the HL7 bodyweight profile has the following cardinality and fixed value constraints:
+
+```
+{
+    "id": "Observation.code.coding:BodyWeightCode.system",
+    "path": "Observation.code.coding.system",
+    ...
+    "min": 1,
+    "max": "1",
+    ...
+    "type": [{
+        "code": "uri"
+    }],
+    "fixedUri": "http://loinc.org",
+     ...
+}
+```
+```
```

Comment:
```suggestion
``` json
```

Reply:
adds nice md formatting in the generated site

Reply:
updated